policy/modules/services/podman.te

```diff
@@ -0,0 +1,265 @@
+fs_getattr_fusefs(podman_user_t)
+fs_mount_fusefs(podman_user_t)
+fs_unmount_fusefs(podman_user_t)
+fs_remount_fusefs(podman_user_t)
+fs_manage_fusefs_dirs(podman_user_t)
+fs_manage_fusefs_files(podman_user_t)
+fs_manage_fusefs_symlinks(podman_user_t)
+fs_exec_fusefs_files(podman_user_t)
+fs_mounton_fusefs(podman_user_t)
+
+kernel_read_fs_sysctls(podman_user_t)
+# to read kernel.unprivileged_userns_clone, if present
+kernel_read_sysctl(podman_user_t)
+
+logging_send_syslog_msg(podman_user_t)
+
+init_write_runtime_socket(podman_user_t)
+
+mount_exec(podman_user_t)
+
+storage_rw_fuse(podman_user_t)
+
+# allow podman to relabel content mounted inside containers
+# when run in rootless mode
+userdom_relabel_generic_user_home_dirs(podman_user_t)
+userdom_relabel_generic_user_home_files(podman_user_t)
+
+ifdef(`init_systemd',`
+	# podman queries the cgroup manager (systemd) over the session bus socket
+	dbus_getattr_session_runtime_socket(podman_user_t)
+	dbus_write_session_runtime_socket(podman_user_t)
+
+	# rootless podman must be able to get login state of the user
+	systemd_dbus_chat_logind(podman_user_t)
+
+	# containers are created as transient user units
+	systemd_start_user_runtime_units(podman_user_t)
+	systemd_stop_user_runtime_units(podman_user_t)
+	systemd_status_user_runtime_units(podman_user_t)
+
+	# podman can read logs from containers which are
+	# sent to the user journal
+	logging_search_logs(podman_user_t)
+	systemd_list_journal_dirs(podman_user_t)
+	systemd_read_journal_files(podman_user_t)
+')
+
+########################################
+#
+# conmon local policy
+#
+
+allow podman_conmon_t self:process signal;
+allow podman_conmon_t self:capability { dac_override dac_read_search sys_ptrace sys_resource };
+allow podman_conmon_t self:cap_userns sys_ptrace;
+allow podman_conmon_t self:fifo_file { rw_fifo_file_perms setattr };
+allow podman_conmon_t self:unix_dgram_socket create_socket_perms;
+dontaudit podman_conmon_t self:capability net_admin;
+
+# conmon will execute crun/runc to create the container
+container_generic_engine_domtrans(podman_conmon_t, podman_t)
+podman_domtrans(podman_conmon_t)
+
+allow podman_conmon_t podman_t:tcp_socket rw_stream_socket_perms;
+allow podman_conmon_t podman_t:unix_stream_socket rw_stream_socket_perms;
+allow podman_conmon_t podman_t:unix_dgram_socket rw_socket_perms;
+ps_process_pattern(podman_conmon_t, podman_t)
+
+domain_use_interactive_fds(podman_conmon_t)
+
+fs_getattr_cgroup(podman_conmon_t)
+fs_search_cgroup_dirs(podman_conmon_t)
+fs_read_cgroup_files(podman_conmon_t)
+fs_watch_cgroup_files(podman_conmon_t)
+
+fs_getattr_tmpfs(podman_conmon_t)
+fs_getattr_xattr_fs(podman_conmon_t)
+
+logging_send_syslog_msg(podman_conmon_t)
+
+miscfiles_read_localization(podman_conmon_t)
+
+userdom_use_user_ptys(podman_conmon_t)
+
+container_read_system_container_state(podman_conmon_t)
+
+# to send/receive data from container ttys
+container_rw_chr_files(podman_conmon_t)
```

Comment:
If the only container chr_files are ttys, then let's have ttys in the name instead of chr_files

Reply:
Other character devices inside the container are also `container_file_t`. That access was granted so that conmon could read/write `/dev/tty` in the container:
```
# ls -lZ /dev/tty
crw-rw-rw-. 1 root root system_u:object_r:container_file_t:s0:c180,c388 5, 0 Jan 20 21:20 /dev/tty
```

Reply:
ok.